lib/plugin-manager.js

```diff
@@ -185,44 +185,73 @@ class PluginManager extends EventEmitter {
   }
 
   /**
+   * Factory function to create instances of source plugins
+   * @param {object} source
+   * @returns {Metadata|S3|Consul}
+   * @private
+   */
+  _sourceFactory(source) {
+    let instance;
+
+    switch (source.type.toLowerCase()) {
+      case 's3':
+
+        // TODO: Support for sources in other buckets
+        instance = new S3(Object.assign(source.parameters, {bucket: this.index.bucket}));
+        break;
+      case 'consul':
+        instance = new Consul({
+          host: Config.get('consul:host'),
+          port: Config.get('consul:port'),
+          secure: Config.get('consul:secure')
+        });
+        break;
+      default:
+        this._error(new Error(`Source type ${source.type} not implemented`));
+    }
+
+    if (instance) {
```

Comment:
This doesn't really do anything. If you don't return a falsey value from `instance`, the call will return undefined anyway